lib/AST/ASTPrinter.cpp

```diff
@@ -1447,6 +1457,40 @@ static unsigned getDepthOfType(Type ty) {
   return depth;
 }
 
+/// Recomputes which inverses must have been written for the given generic
+/// signature.
+static void reconstituteInverses(GenericSignature genericSig,
+                                 ArrayRef<Type> genericParams,
```

Comment:
these are not just generic parameters, they can also be dependent member types `Self.A` or a protocol, so perhaps typeParams or defaultedTypeParams is a better name.